akka-http/src/main/java/akka/http/javadsl/coding/Coder.java

```diff
@@ -25,9 +29,40 @@
         this.underlying = underlying;
     }
 
+    public HttpResponse encodeMessage(HttpResponse message) {
+        boolean shouldApply = (boolean) underlying.messageFilter().apply((HttpMessage) message);
```

Comment:
`JavaMappings` is not so convenient to be used in Java code, so I fell back to the un-fancy cast.